lib/ansible/plugins/inventory/kubevirt.py

```diff
@@ -0,0 +1,219 @@
+                variable.
+              type: str
+          username:
+              description:
+              - Provide a username for authenticating with the API. Can also be specified via K8S_AUTH_USERNAME
+                environment variable.
+              type: str
+          password:
+              description:
+              - Provide a password for authenticating with the API. Can also be specified via K8S_AUTH_PASSWORD
+                environment variable.
+              type: str
+          cert_file:
+              description:
+              - Path to a certificate used to authenticate with the API. Can also be specified via K8S_AUTH_CERT_FILE
+                environment variable.
+              type: str
+          key_file:
+              description:
+              - Path to a key file used to authenticate with the API. Can also be specified via K8S_AUTH_HOST
+                environment variable.
+              type: str
+          ssl_ca_cert:
+              description:
+              - Path to a CA certificate used to authenticate with the API. Can also be specified via
+                K8S_AUTH_SSL_CA_CERT environment variable.
+              type: str
+          verify_ssl:
+              description:
+              - "Whether or not to verify the API server's SSL certificates. Can also be specified via
+                K8S_AUTH_VERIFY_SSL environment variable."
+              type: bool
+          namespaces:
+              description:
+              - List of namespaces. If not specified, will fetch all virtual machines for all namespaces user is authorized
+                to access.
+              type: list
+          network_name:
+              description:
+              - In case of multiple network attached to virtual machine, define which interface should be returned as primary IP
+                address.
+              type: str
+          api_version:
+              description:
+              - "Specify the KubeVirt API version."
+              type: str
+    requirements:
+    - "openshift >= 0.6"
+    - "PyYAML >= 3.11"
+'''
+
+EXAMPLES = '''
+# File must be named kubevirt.yaml or kubevirt.yml
+
+# Authenticate with token, and return all virtual machines for all namespaces
+plugin: kubevirt
+connections:
+ - host: https://kubevirt.io
+   token: xxxxxxxxxxxxxxxx
+   ssl_verify: false
+
+# Use default config (~/.kube/config) file and active context, and return vms with interfaces
+# connected to network myovsnetwork and from namespace vms
+plugin: kubevirt
+connections:
+  - namespaces:
+      - vms
+    network_name: myovsnetwork
+'''
+
+from ansible.plugins.inventory.k8s import K8sInventoryException, InventoryModule as K8sInventoryModule, format_dynamic_api_exc
+
+try:
+    from openshift.dynamic.exceptions import DynamicApiError
+except ImportError:
+    pass
+
+
+API_VERSION = 'kubevirt.io/v1alpha3'
+
+
+class InventoryModule(K8sInventoryModule):
```

Comment:
you are also missing doc fragments for classes using constructable and cacheable

Reply:
This is done as part of inheritance from `K8sInventoryModule`
`
lib/ansible/plugins/inventory/k8s.py:class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable, K8sAnsibleMixin):
`

Reply:
@machacekondra still, it requires the docs

also that module would not have passed review, it got merged w/o it so it is not a good example to follow

Reply:
It's addressed here https://github.com/ansible/ansible/pull/47803/files#diff-4e0a09d18f6dfe19eb9d0b30c0a53bafR16